frontend/components/Page.js

```diff
@@ -0,0 +1,66 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { string, object, bool } from 'prop-types'
+
+import Head from './Head'
+import NavBar from './NavBar'
+import { Container, Dimmer, Loader } from 'semantic-ui-react'
+import AuthProvider from '../contexts/AuthContext'
+import { useRouter } from 'next/router'
+
+const Content = ({ reqSignIn, reqSignOut, children }) => {
+  const { push } = useRouter()
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    const isAuthenticated = window.session?.user !== null
+
+    if (reqSignIn && !isAuthenticated) {
+      push('/login').then()
+    } else if (reqSignOut && isAuthenticated) {
+      push('/').then()
+    } else {
+      setLoading(false)
+    }
+  }, [loading])
+
+  if (loading) {
+    return (
+      <Dimmer active inverted>
+        <Loader>Loading...</Loader>
+      </Dimmer>
+    )
+  } else {
+    return <Container style={{ marginTop: 30 }}>{children}</Container>
+  }
+}
+
+export const Page = props => {
+  return (
+    <AuthProvider>
+      <Head
+        description={props.head?.description}
+        ogImage={props.head?.ogImage}
+        title={props.head?.title}
+        url={props.head?.url}
```

Comment:
I think this is worse interface than before. 

1. Using an object as a prop is a bad practice since you can change a field and it will still be seen as equal
2. Being able to specify things on a page using `<Head>` seems much nicer than passing in an object to our `<Page>` as a prop.

Reply:
1. Using an object as a prop is a bad practice since you can change a field and it will still be seen as equal

Shouldn't the head content change between different renders of the same page?!

> 2. Being able to specify things on a page using `<Head>` seems much nicer than passing in an object to our `<Page>` as a prop.

Ok, We can remove it from the `Page` and add it as a child component like other components inside `Page`.

Reply:
It's not really an issue in this case, as it is, but when you are designing interfaces on React components we should just avoid having `object`  (including `Array`) as a prop. The reasoning is that they are mutable but will continue to pass equality with `===`  when changed. When you try and optimize components with these mutable `object`s as props, it's a pain. 

> Ok, We can remove it from the Page and add it as a child component like other components inside Page.

Great, yeah. Is it actually possible to use `Head` multiple times in the same page so we can still have some defaults?